src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -299,24 +299,57 @@ namespace OpenLoco::World::MapGenerator
         applySurfaceStyleToMarkedTiles(heightMap, surfaceStyle, false);
     }
 
+    // 0x0046A0D8, 0x0046A227
+    static void generateTerrainInRandomAreas(uint8_t surfaceStyle, uint16_t minTiles, uint16_t maxTiles)
+    {
+        auto ecx = getGameState().rng.randNext(80, 143);
+        for (; ecx > 0; ecx--)
```

Comment:
we could restructure these into normal for loops as well. Not sure what you think. 
`for(auto i = 0U; i < numberOfAreas; ++i)...`
`for(auto j = 0U; j < numberOfTiles; ++j)...`